contracts/task-queue/src/tests/common.rs

```diff
@@ -455,6 +466,69 @@ where
     );
 }
 
+pub fn mock_hook_consumer_test<C>(chain: C, mock_consumer: MockHookConsumerContract<C>)
+where
+    C: CwEnv + AltSigner,
+    C::Sender: Addressable,
+{
+    let coin = Coin {
+        denom: DENOM.to_string(),
+        amount: Uint128::one(),
+    };
+    let funds = vec![coin.clone()];
+
+    // Setup the task contract
+    let verifier = chain.alt_signer(VERIFIER_INDEX);
+    let msg = InstantiateMsg {
+        requestor: Requestor::OpenPayment(coin),
+        timeout: mock_timeout(200),
+        verifier: verifier.addr().into(),
+    };
+    let task_contract = setup(chain.clone(), msg);
+
+    // Establish hooks
+    mock_consumer
+        .add_hooks(task_contract.addr_str().unwrap())
+        .unwrap();
+
+    // Ensure created counter starts at 0
+    let counter = mock_consumer.created_count().unwrap();
+    assert!(counter.is_zero());
+
+    // Create a task
+    let payload = json!({"x": 5});
+    let task_id = make_task_with_funds(&task_contract, "Test Task", None, &payload, &funds);
+
+    // Verify task created hook
+    let counter = mock_consumer.created_count().unwrap();
+    assert_eq!(counter, Uint128::one());
```

Comment:
Nitpicking, but seems like a regular `u128` for a counter would be easier to use and read.